boards/posix/nrf52_bsim/irq_handler.c

```diff
@@ -390,3 +390,8 @@ void __SEV(void)
 {
 	CPU_event_set_flag = true;
 }
+
+/* Model ARM Data Synchronization Barrier instruction as no-op */
+void __DSB(void)
```

Comment:
The no-op should be sufficient for nrf52_bsim. __DMB is defined in the HW model (see https://github.com/BabbleSim/ext_NRF52_hw_models/blob/master/src/nrfx/hal/nrf_soc_if.h#L98) . For consistency, I would prefer __DSB to come from there too.

Reply:
I agree with @wopu-ot that this one does not belong here (the `__WFE` `__WFI` are here because of the model of the event behaviour)
But I'm thinking that maybe the `__DMB` and `__NOP` should not have been placed in the HW models, but in a new `boards/posix/nrf52_bsim/cmsis.h` as macros, and have this `cmsis.h` be included from `board_soc.h`. (This __<asm_inst> macros/functions are defined by CMSIS)

@wopu-ot what do you think about also moving the DMB and NOP to a new cmsis.h?

Reply:
@aescolar @wopu-ot thanks for your input on this. I would probably prefer to see the functions in some cmsis.h as macros or static inline empty functions, what do you think?

Reply:
@aescolar @ioannisg I would be fine with moving DMB and NOP, my point was more that DSB should be next to DMB.

Reply:
@ioannisg Would you then create that new file with these definitions, and send a PR to the HW models removing those 2 static inlines from there?
@rugeGerritsen I guess you are ok with __DMB and __NOP being in that new cmsis.h?

Reply:
@aescolar . I'm fine with moving those to cmsis.h

Reply:
@aescolar do get this PR pass CI, I suggest I do this for DSB and we can move the DMP, NOP, in the cmisis (and remove from BabbleSIm) in a subsequent PR, what do you think?

Reply:
@ioannisg it can be. Would it be you the kind person preparing that 2nd PR? :)

Reply:
sure, can do that

Reply:
@aescolar @wopu-ot https://github.com/BabbleSim/ext_NRF52_hw_models/pull/6
I guess, if you merge that one, then we can send a PR in Zephyr to add these functions in cmsis.h